packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -117,6 +99,61 @@ export const getLetterEditingOptions = <TDate>(
   }
 };
 
+export const cleanDigitSectionValue = <TDate>(
+  utils: MuiPickersAdapter<TDate>,
+  value: number,
+  sectionType: FieldSectionType,
+  format: string,
+  hasLeadingZeros: boolean,
+  sectionBoundaries: FieldSectionValueBoundaries<TDate, any>,
+) => {
+  const hasLetterPrefix = () => {
+    const startOfYear = utils.startOfYear(utils.date()!);
+    const startOfYearStr = utils.formatByString(startOfYear, format);
+
+    return Number.isNaN(Number(startOfYearStr));
+  };
+
+  if (process.env.NODE_ENV !== 'production') {
```

Comment:
Should never happen since we are the one adding the supported tokens to the adapter.